### ISSUE TITLE
CommerceRuntime sample fails when Nuget reference to Microsoft.Azure.Cosmos is added

### ISSUE THREAD

**Issue by MaxWilson** [microsoft/Dynamics365Commerce.ScaleUnit#8]
Consuming Microsoft.Azure.Cosmos leads to exceptions at build time from both the proxy generator and the ScaleUnit package builder. Either this is a documentation hole (if there's a workaround) or a product bug (if there isn't). **Edit:** It turns out this is a regression from 9.32.

Minimal repro:

1.) Clone the repo.
2.) Check out branch 9.34. (Does not repro on 9.32.)
3.) In src\ScaleUnitSample\CommerceRuntime\CommerceRuntime.csproj, add a reference to Microsoft.Azure.Cosmos. I don't think the version matters but I used 
```
<PackageReference Include="Microsoft.Azure.Cosmos" Version="3.23.0" />
```
4.) Add a reference somewhere to a type from Microsoft.Azure.Cosmos, e.g. I added to BoundController.cs 
```
    public class BoundController : IController
    {
        private Microsoft.Azure.Cosmos.CosmosClient stub = null; // add a reference to a type from Microsoft.Azure.Cosmos 
                                                                    // so the assembly won't be ignored by the compiler
        ...
```
5.) Build the solution.

**EXPECTED:** everything builds correctly, ScaleUnit package is created.
**ACTUAL:** strange error from proxy generator

```
"C:\Repo\Dynamics365Commerce.ScaleUnit\src\ScaleUnitSample\POS\POS.csproj" (default target) (4:2) ->
(RunCommerceTypeScriptExtensionsProxyGenerator target) ->
  EXEC : Commerce Proxy Generator failed due to the error : Method not found: 'Microsoft.AspNet.OData.Builder.LengthPro
pertyConfiguration Microsoft.AspNet.OData.Builder.StructuralTypeConfiguration`1.Property(System.Linq.Expressions.Expres
sion`1<System.Func`2<!0,System.String>>)'.. [C:\Repo\Dynamics365Commerce.ScaleUnit\src\ScaleUnitSample\POS\POS.csproj]
  C:\Users\Admin1e20ff3e45\.nuget\packages\microsoft.dynamics.commerce.sdk.pos\9.34.22047.3\Dependencies\CoreProxyGener
ator\build\Microsoft.Dynamics.Commerce.Tools.CoreProxyGenerator.props(126,5): error MSB3073: The command ""C:\Users\Adm
in1e20ff3e45\.nuget\packages\microsoft.dynamics.commerce.sdk.pos\9.34.22047.3\Dependencies\CoreProxyGenerator\build\..\
tools\CommerceProxyGenerator.exe" "C:\Users\Admin1e20ff3e45\.nuget\packages\microsoft.dynamics.commerce.sdk.pos\9.34.22
047.3\Dependencies\ExtensionsProxyGenerator\build\\..\tools\Microsoft.Dynamics.Retail.RetailServerLibrary.AspNetCore.dl
l" "C:\Repo\Dynamics365Commerce.ScaleUnit\src\ScaleUnitSample\CommerceRuntime\bin\Debug\netstandard2.0\CommerceRuntime.
dll" /a:TypeScriptExtensions /outputpath:obj\Debug\" exited with code 1. [C:\Repo\Dynamics365Commerce.ScaleUnit\src\Sca
leUnitSample\POS\POS.csproj]

    0 Warning(s)
    2 Error(s)
```

**ADDITIONALLY**
If I edit ScaleUnit\ScaleUnit.csproj to remove the references to POS and ChannelDatabase, to work around the proxy generation bug, I get an error about assembly manifests during package generation:

```
Build FAILED.

"C:\Repo\Dynamics365Commerce.ScaleUnit\src\ScaleUnitSample\ScaleUnit\ScaleUnit.csproj" (default target) (1) ->
(GetUpstreamPackagingOutput target) ->
  C:\Users\Admin1e20ff3e45\.nuget\packages\microsoft.dynamics.commerce.sdk.scaleunit\9.34.22047.3\build\Microsoft.Dynam
ics.Commerce.Sdk.ScaleUnit.targets(62,5): error : Could not load file or assembly 'Cosmos.CRTCompat.dll' or one of its
dependencies. The module was expected to contain an assembly manifest. [C:\Repo\Dynamics365Commerce.ScaleUnit\src\Scale
UnitSample\ScaleUnit\ScaleUnit.csproj]

    0 Warning(s)
    1 Error(s)
```

**REQUEST:** guidance on how to consume Microsoft.Azure.Cosmos and assemblies like it.

**KNOWN WORKAROUNDS**: Use branch 9.32 instead.

**Comment by iedeny** [comment]
Issue is fixed and fix will be cherry-picked to 9.34+ releases

**Comment by MaxWilson** [comment]
Fixed!
